src/library/components/stem-finder-result.js

```diff
@@ -125,36 +126,60 @@ var StemFinderResult = Component({
+    const divClass = 'portal-pages-finder-result-favorite' + active
+    return (
+      <div className={divClass} onClick={this.toggleFavorite}>
+        <i className={'icon-favorite'}></i>
+      </div>
     )
   },
 
   render: function () {
-    var resource = this.props.resource
-    var options = {href: resource.stem_resource_url}
+    const resource = this.props.resource
+
+    // truncate title and/or description if they are too long for resource card height
+    const maxCharTitle = 125
+    const maxCharDesc = 320
+    let resourceName = portalObjectHelpers.shortenText(resource.name, maxCharTitle, true)
+    let shortDesc = resource.filteredShortDescription
+    if (shortDesc.length + resource.name.length >= maxCharDesc) { // use full resource name on 'back' of card, not truncated version
+      shortDesc = portalObjectHelpers.shortenText(shortDesc, maxCharDesc - resource.name.length, true)
+    }
 
     if (this.state.hovering || this.state.lightbox) {
-      return div({className: 'portal-pages-finder-result col-4', onClick: this.toggleLightbox, onMouseOver: this.handleMouseOver, onMouseOut: this.handleMouseOut},
-        a(options,
-          div({className: 'portal-pages-finder-result-description'}, resource.filteredShortDescription),
-          this.renderFavoriteStar()
-        ),
-        GradeLevels({resource: resource}),
-        this.renderFavoriteStar()
+      return (
+        <div className={'portal-pages-finder-result col-4'} onClick={this.toggleLightbox} onMouseOver={this.handleMouseOver} onMouseOut={this.handleMouseOut}>
+          <a href={resource.stem_resource_url}>
+            <div className={'portal-pages-finder-result-description'}>
+              <div className={'title'}>
+                {resource.name}
+              </div>
+              <div>
+                {shortDesc}
+              </div>
+            </div>
+            {this.renderFavoriteStar()}
+          </a>
+          {GradeLevels({resource: resource})}
+          {this.renderFavoriteStar()}
+        </div>
       )
     }
-    return div({className: 'portal-pages-finder-result col-4', onClick: this.toggleLightbox, onMouseOver: this.handleMouseOver, onMouseOut: this.handleMouseOut},
-      a(options,
-        div({className: 'portal-pages-finder-result-image-preview'},
-          img({alt: resource.name, src: resource.icon.url}),
-          ResourceType({resource: resource})
-        ),
-        div({className: 'portal-pages-finder-result-name'}, resource.name),
-        this.renderFavoriteStar()
-      ),
-      GradeLevels({resource: resource})
+    return (
+      <div className={'portal-pages-finder-result col-4'} onClick={this.toggleLightbox} onMouseOver={this.handleMouseOver} onMouseOut={this.handleMouseOut}>
+        <a href={resource.stem_resource_url}>
+          <div className={'portal-pages-finder-result-image-preview'}>
+            <img alt={resource.name} src={resource.icon.url} />
+            {ResourceType({resource: resource})}
```

Comment:
Since this is a component you can use this slightly nicer syntax:
`<ResourceType resource={resource} />`
@pjanik does using this approach as other benefits besides looking nicer?

Reply:
@scytacki, perhaps not, the whole JSX thing is about looking nicer and/or more readable. 😉 
In this case, an additional benefit is that it's clear that `ResourceType` is a React component. Otherwise, it looks pretty strange (could be a function).